### DISCUSSION TITLE
Request for "accept next word" keybinding for IntelliJ

### DISCUSSION THREAD

**Discussion by mikehearn** [community/community#65179] [Copilot Conversations] [answered]
### Select Topic Area

Product Feedback

### Body

In IntelliJ to accept a suggestion you must press tab, which then accepts all of it. But often half the suggestion is right and the rest is wrong. I'd like much finer grained acceptance so I can accept up to the bit where it goes wrong, then continue typing from there.

**Comment by Ket-0626** [comment]
To achieve finer-grained acceptance of code completion suggestions in IntelliJ, you can try the following options:

1. Use the arrow keys to navigate through the suggestion list and select the desired suggestion. Press Enter to accept the selected suggestion[2].

2. Use a plugin like "Smart Keys" to customize the behavior of the Tab key. This plugin allows you to map the Tab key to different actions depending on the context. For example, you can map Tab to "accept suggestion" only when the suggestion is fully correct, and map it to "indent" or "autocomplete" in other cases[1].

3. Configure IntelliJ to use a different key for accepting suggestions. For example, you can map the Enter key to "newline" instead of "accept suggestion", and use a different key (e.g. Ctrl+Space) to trigger code completion[1].

4. Configure IntelliJ to show a popup for each item in the suggestion list with the documentation for the class, method, or field currently highlighted in the lookup list. This can help you identify which suggestion is the most appropriate[2].

5. Configure IntelliJ to insert parentheses automatically when applicable. This can save you time when completing function/method calls[2].

6. Configure IntelliJ to show parameter name hints on completion. This can help you remember the names of the parameters for a function/method[2].

7. Use the arrow keys to navigate to the point where the suggestion goes wrong, and continue typing from there. This allows you to accept part of the suggestion and modify the rest[2].

Note that some of these options may require additional configuration or the installation of plugins. You can refer to the IntelliJ documentation or community forums for more information.

**Reply by mikehearn** [reply]
Is this answer generated by an AI? None of these responses are useful and the claims have reference numbers but no references are given. 

Please don't copy/paste from ChatGPT like that. This was a request for a feature to be added, not a generic plea for help. The plugin doesn't let you do what I asked currently.

**Comment by Akash1134** [comment]
Hi @Ket-0626 ,

While we encourage non-AI answers to questions in community, we also love experimenting and using new technologies. We ask if you do use ChatGPT or similar programs, please disclose what software and model version number you used to generate the response you used in your reply in order to promote transparency in the community.

ChatGPT answers in a tone that sounds like a tech support professional and while we support trying out this software, it is important to verify that they are solutions before posting otherwise it may mislead users into thinking that they are receiving official responses when they are not.

Per ChatGPT's website, the program does have limitations like:
- "May occasionally generate incorrect information"
- "May occasionally produce harmful instructions"
- "Limited knowledge of world and events after 2021"

**Comment by Akash1134** [comment, marked as answer]
Hi @mikehearn ,

Thank you for your feedback. We will share it with our teams internally for further investigation. 

Please note that we can't guarantee that every bit of feedback sees action, but we are paying attention.

**Comment by informatik01** [comment]
I am using GoLand (IntelliJ IDEA should be the same in this regard) with GitHub Copilot on macOS Sequoia and using **⌘ →** gets the job done: suggestions are getting accepted in a **word by word** manner - just what I need.
